### PR TITLE
Drag and drop in the navigator

### PR DESCRIPTION
**Pull request by alexzherdev**

### CODE REVIEW

File: web/src/scripts/components/filetree/Node.js [hunk 1]
```diff
@@ -137,27 +175,34 @@ export default class Node extends Component {
...
+          {this.renderButton()}
         </div>
-        {this.renderButton()}
-      </div>
+      ))
     )
   }
 }
+
+export default flow(
```
Comment (dabbott) [inline web/src/scripts/components/filetree/Node.js]: Can we use decorators for these? I don't think we use this `flow` pattern anywhere else

File: web/src/scripts/components/filetree/Node.js [hunk 2]
```diff
@@ -137,27 +175,34 @@ export default class Node extends Component {
...
-      >
-        {isDirectory(type) && (
-          <NodeCaret expanded={expanded}/>
-        )}
-        <div style={isDirectory(type) ? styles.folderIcon : styles.fileIcon} />
-        <div className={'flex-variable'} style={styles.nodeText}>
-          {name}
+      connectDragSource(connectDropTarget(
+        <div
+          style={styles.getNodeStyle(depth, selected, hover || isDirectory(type) && isDraggingOver)}
```
Comment (dabbott) [inline web/src/scripts/components/filetree/Node.js]: Very minor, but I'm thinking we should give these conditions a name for clarity, e.g:

```javascript
const highlightDropTarget = isDirectory(type) && isDraggingOver
```

File: web/src/scripts/containers/ProjectNavigator.jsx [hunk 3]
```diff
@@ -117,6 +117,13 @@ class ProjectNavigator extends Component {
     textEditorCompositeActions.updateDecoEntryRequire(requireText)
   }
 
+  handleDrop = (sourceNode, targetNode) => {
+    this.props.fileTreeCompositeActions.renameFile(
```
Comment (dabbott) [inline web/src/scripts/containers/ProjectNavigator.jsx]: We probably want to call either `renameFile` or `renameDir` depending on what kind of `sourceNode` we're handling

Comment (alexzherdev) [inline web/src/scripts/containers/ProjectNavigator.jsx]: Interestingly, it seemed to work with dirs using just `renameFile`. But of course I'll update any way.

**Pull request merged**